[{"count": 0, "attachment_id": null, "bug_id": 6717, "text": "Hi there,\nI'm using your Tomcat3.3a version.\nI wrote this Servlet:\npublic class TestRequestHeadersServlet extends HttpServlet \n{\n\tprotected void service(HttpServletRequest req, HttpServletResponse resp)\n                throws ServletException, java.io.IOException\n\t{\n\t\tEnumeration headersNames = req.getHeaderNames();\n\t\t\n\t\twhile ( headersNames.hasMoreElements() )\n\t\t{\n\t\t\tString headerName = (String)headersNames.nextElement();\n\t\t\tString headerValue = req.getHeader( headerName );\n\t\t\tSystem.out.println( headerName + \":\" + headerValue );\n\t\t}\n\t} \n}\n\nwhich basically, as I understand, prints all client Http headers.\nit works fine in most cases but when I'm using a client that sends these \nheaders:\nx-up-uplink:pcwap1.cellcom.co.il\nx-up-fax-limit:100000\nx-up-subno:1010498229-88004_pcwap1.cellcom.co.il\nx-up-fax-accepts:text/plain, application/postscript, application/msword, \napplication/rtf, application/pdf\nhost:212.179.27.151:8080\nAccept-Language:en\nUser-Agent:SIE-S45/14 UP.Browser/5.0.1.1.102 (GUI) UP.Link/4.3.2\nx-up-devcap-max-pdu:2800\nx-up-devcap-charset:utf-8\nConnection:Keep-Alive\nEncoding-version:1.3\nx-up-devcap-iscolor:0\nx-up-fax-encodings:7bit, 8bit, base64, quoted-printable\nAccept:application/vnd.wap.wmlc;Type=4365, application/vnd.wap.wmlc;Type=4360, \napplication/vnd.wap.wmlc;Type=1108, application/vnd.wap.wmlc;Level=1.3, \napplication/vnd.wap.wmlc, application/vnd.wap.wmlscriptc, \napplication/vnd.wap.sic, application/vnd.wap.slc, application/vnd.wap.coc, \napplication/vnd.uplanet.signal, application/vnd.uplanet.cacheop-wbxml, \napplication/vnd.uplanet.alert-wbxml, application/vnd.uplanet.channel-wbxml, \napplication/vnd.uplanet.list-wbxml, application/vnd.uplanet.listcmd-wbxml, \napplication/vnd.uplanet.bearer-choice-wbxml, \napplication/vnd.wap.multipart.related, application/vnd.wap.multipart.mixed, \napplication/vnd.syncml-xml-wbxml, application/x-up-device, \napplication/vnd.phonecom.mmc-wbxml;Type=4364, application/vnd.phonecom.mmc-\nwbxml, application/vnd.phonecom.im, application/octet-stream, \nimage/vnd.wap.wbmp, image/bmp, image/x-MS-bmp, audio/midi, audio/x-midi, \naudio/mid, text/x-vCalendar, text/x-vCard, application/x-\nmmc.wallpaper;type=image/bmp;size=2000;gray=1;w=101;h=64, application/x-\nmmc.ringtone;type=audio/midi;size=2000;voices=1, application/x-\nmmc.title;charset=us-ascii;size=30, \nimage/bmp,image/vnd.wap.wbmp,application/vnd.uplanet.alert,application/x-up-\nalert,application/vnd.uplanet.bearer-\nchoice,application/vnd.uplanet.cacheop,application/x-up-\ncacheop,application/vnd.uplanet.channel,application/vnd.uplanet.list,application\n/vnd.uplanet.listcmd,application/vnd.phonecom.mmc-xml,text/x-\nwap.wml,text/vnd.wap.wml,text/x-hdml,text/html,text/vnd.wap.wmlscript\nx-up-devcap-numsoftkeys:2\nx-up-devcap-gui:1\nx-up-devcap-softkeysize:5\nAccept-Charset:utf-8, UTF-8,\n\non the first two times i get this kind of response:\nx-up-uplink:pcwap1.cellcom.co.il\nx-up-fax-limit:100000\nx-up-subno:1010498229-88004_pcwap1.cellcom.co.il\nx-up-fax-accepts:text/plain, application/postscript, application/msword, \napplication/rtf, application/pdf\nhost:212.179.27.151:8080\nAccept-Language:en\nUser-Agent:SIE-S45/14 UP.Browser/5.0.1.1.102 (GUI) UP.Link/4.3.2\nx-up-devcap-max-pdu:2800\nx-up-devcap-charset:utf-8\nConnection:Keep-Alive\nEncoding-version:1.3\nx-up-devcap-iscolor:0\nx-up-fax-encodings:7bit, 8bit, base64, quoted-printable\n2002-02-27 15:33:12 - Ctx() : Exception in R(  \n+ /servlet/TestRequestHeadersServlet + null) - \njava.lang.StringIndexOutOfBoundsException: length must be >= 0\n        at java.lang.String.<init>(String.java:362)\n        at java.lang.String.<init>(String.java:411)\n        at org.apache.tomcat.util.buf.ByteChunk.toString(Unknown Source)\n        at org.apache.tomcat.util.buf.MessageBytes.toString(Unknown Source)\n        at org.apache.tomcat.util.http.MimeHeaders.getHeader(Unknown Source)\n        at org.apache.tomcat.core.Request.getHeader(Unknown Source)\n        at org.apache.tomcat.facade.HttpServletRequestFacade.getHeader(Unknown \nSource)\n        at TestRequestHeadersServlet.service(TestRequestHeadersServlet.java:19)\n        at javax.servlet.http.HttpServlet.service(HttpServlet.java)\n        at org.apache.tomcat.facade.ServletHandler.doService(Unknown Source)\n        at org.apache.tomcat.core.Handler.invoke(Unknown Source)\n        at org.apache.tomcat.core.Handler.service(Unknown Source)\n        at org.apache.tomcat.facade.ServletHandler.service(Unknown Source)\n        at org.apache.tomcat.core.ContextManager.internalService(Unknown Source)\n        at org.apache.tomcat.core.ContextManager.service(Unknown Source)\n        at org.apache.tomcat.modules.server.Http10Interceptor.processConnection\n(Unknown Source)\n        at org.apache.tomcat.util.net.TcpWorkerThread.runIt(Unknown Source)\n        at org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run\n(Unknown Source)\n        at java.lang.Thread.run(Thread.java:484)\n-----------------------------------------------------\nand after that it seems like working fine.\n\nI'd appreciate any response.\n\nThanks,\nOfer", "id": 11213, "time": "2002-02-27T13:22:29Z", "creator": "ofer@feathersys.com", "creation_time": "2002-02-27T13:22:29Z", "tags": [], "is_private": false}, {"count": 1, "tags": [], "creator": "william.barker@wilshire.com", "attachment_id": null, "is_private": false, "id": 11275, "time": "2002-02-28T03:23:48Z", "bug_id": 6717, "creation_time": "2002-02-28T03:23:48Z", "text": "I think that I've tracked this down, so I'm marking it as FIXED.  Since this \none is hard to reproduce (without your client), it would be helpful if you \ncould test against the 2/28 nightly."}]